### DISCUSSION TITLE
Possibility of volume balancing?

### DISCUSSION THREAD

**Discussion by aschenkuttel** [lavalink-devs/Lavalink#963] [Q&A] [answered]
Hey, 
I just wanted to know if there's anything in plan or if it's even possible to balance the volumes of tracks. A lot of youtube videos are pretty different in regards of the base volume and it would be nice to have some sort of balancer which adjusts an internal volume setting based on the current track. Is it possible to estimate this from beginning frames or even the whole track? I'm not versed in java so I'm not sure whats possible.

Or would that be something the libraries should handle?

Cheers :)

**Comment by topi314** [comment, marked as answer]
Normalizing audio is kinda tricky since for the best result you would need to download the whole track & process it as a whole.
A different option would be just keep down the peaks?

Anyway, lavalink itself has no plans to implement such a thing as of right now & probably not in the future.
This could be implemented as a filter & lavalink plugin, which does require JVM knowledge to build.

This also isn't really something libraries can tackle, since they don't receive the audio

**Reply by aschenkuttel** [reply]
I probably won't be able to do the filter myself but do you have any documentation for that to push me in the right direction?

**Reply by topi314** [reply]
https://lavalink.dev/api/plugins.html
